### PR TITLE
feat: lesson14

### PR DESCRIPTION
**Pull request by nakagawamai**
# [Issue14](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#14)
13で作ったモーダル内にinput (typeはnumber)をおいて、クリックした際にinput(type number)のvalueを取得して、リクエストできるようにしてください。fetchを直接ラップしている関数の引数にその値を渡してください

## Uneasy part in implementation
- I'm worried if the part that passes the value to the argument of the function that wraps fetch directly is done well.

### CODE REVIEW

File: lesson14/script.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        console.error(error);
+    }finally{
+        removeLoadingImg();
+    }
+}
+
+const getInputNumber = () => {
+    const inputNumber = document.getElementById('js-input-number');
+    return inputNumber.value;
+}
```
Comment (ghost) [inline lesson14/script.js]: IMO
If you only use the input value, I think you should just return that, what do you think?
```suggestion
const getInputNumberValue = () => document.getElementById("js-input-number").value;
```

File: lesson14/script.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    const loadingImg = document.createElement('img');
+    div.id           = "js-loading";
+    loadingImg.src   = "img/loading-circle.gif";
+    loadingImg.alt   = "ローディング画像";
+    document.body.insertAdjacentElement('afterbegin',div).appendChild(loadingImg);
+}
+
+const removeLoadingImg = () => {
+    document.getElementById('js-loading').remove();
+}
```
Comment (ghost) [inline lesson14/script.js]: IMO
You omitted the parentheses in the MODAL function. Let's align here as well.
```suggestion
const removeLoadingImg = () => document.getElementById('js-loading').remove();
```

File: lesson14/script.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
+const showLoadingImg = () => {
+    const div        = document.createElement('div');
```
Comment (seeeen044) [inline lesson14/script.js]: IMO
It is better to have specific names for variables. For example, "loadingPlace".

File: lesson14/script.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        a.href        = `/${list.a}`;
+        a.textContent = list.text;
+        img.src       = list.img;
+        img.alt       = list.alt;
+        fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
+    } 
+    document.getElementById('js-list').appendChild(fragment);
+}
+
+const fetchdata = async () => {
```
Comment (seeeen044) [inline lesson14/script.js]:
```suggestion
const fetchData = async () => {
```
nits
It is very detailed but should be consistent with the Camel case.

File: lesson14/script.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+const init = async (number) => {
+    showLoadingImg();
+    try{
+        console.log(number);
+        const result = await fetchdata();
+        if(!result){
+            return
+        }
```
Comment (yasu-pro) [inline lesson14/script.js]: I have learned a lot and I am happy to comment on this.
I was not using the if statement here and was getting a synctax error in the console if the value was false.
I learned how to work around it by looking at your code

File: lesson14/script.js (hunk 1 above)
Comment (nakagawamai) [reply lesson14/script.js]: Thank you for your comment!
I think so too.
I fixed the part that can be omitted.
https://github.com/nakagawamai/js-lesson/pull/16/commits/507f100d13798d78da9ce5d6355d49b3483a14f1

File: lesson14/script.js (hunk 2 above)
Comment (nakagawamai) [reply lesson14/script.js]: Certainly, this should be the case here as well.
I fixed.
https://github.com/nakagawamai/js-lesson/commit/507f100d13798d78da9ce5d6355d49b3483a14f1

File: lesson14/script.js (hunk 3 above)
Comment (nakagawamai) [reply lesson14/script.js]: Thank you for your comment!
I renamed  the variables.
https://github.com/nakagawamai/js-lesson/pull/16/commits/c9329c8c192860a0856fb4ccb32bbca28c07c96b

File: lesson14/script.js (hunk 4 above)
Comment (nakagawamai) [reply lesson14/script.js]: Thank you for teaching me!
I changed to the Camel case.
https://github.com/nakagawamai/js-lesson/pull/16/commits/5e9cffb4b66a9a43a4f3b5e7613cc2126382ec23

File: lesson14/script.js (hunk 5 above)
Comment (nakagawamai) [reply lesson14/script.js]: Thank you for your comment!
This method was taught by Monaka-san!

**Review by seeeen044 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**